### PR TITLE
Global Ratelimit tests CEPH-83574915, CEPH-83574916

### PR DESCRIPTION
**Pull request by TejasC88**
Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/s3cmd_bucket_stats.txt
                  http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/ratelimit_global.txt

Also fix for s3cmd bucket_stats failure seen in earlier runs

### CODE REVIEW

**Review by viduship — APPROVED** [review]

File: rgw/v2/tests/s3cmd/test_bucket_stats.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def test_exec(config, ssh_con):
             log.info(f"Bucket {bucket_name} created")
             utils.exec_shell_cmd(f"fallocate -l 25m obj25m")
             object_name = (
-                f"s3://{bucket_name}/encyclopedia/space & universe/.bkp/journal$i"
+                f"s3://{bucket_name}/encyclopedia/space.and.universe/.bkp/journal$i"
             )
```
Comment (anrao19) [inline rgw/v2/tests/s3cmd/test_bucket_stats.py:64]: @TejasC88 object name : f"s3://{bucket_name}/encyclopedia/space & universe/.bkp/journal$i" is expected as per BZ , this is a custer BZ
and error observed in log is expected

Comment (TejasC88) [reply rgw/v2/tests/s3cmd/test_bucket_stats.py:64]: thanks @anrao19 for catching this , I had not gone through the BZ , the fix is missing in 6.0 , have cloned the 5.x BZ for 6.x : https://bugzilla.redhat.com/show_bug.cgi?id=2181424

Comment (TejasC88) [reply rgw/v2/tests/s3cmd/test_bucket_stats.py]: Have reverted  my changes , and here is the failure log for the stats script:
http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/bucket_stats_s3cmd.txt

Comment (viduship) [reply rgw/v2/tests/s3cmd/test_bucket_stats.py]: @anrao19 , hey, was this bug not fixed, is it a regression?

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+    s3cmd_reusable.rate_limit_read(bucket_name, max_read_ops, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    s3cmd_reusable.rate_limit_write(bucket_name, max_write_ops, ssl)
+
+    # sleep a minute to reset the ops limit
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
```
Comment (mkasturi18) [inline rgw/v2/tests/s3cmd/test_ratelimit_global.py:117]: @TejasC88 , can we add the sleep duration in the method rate_limit_read rate_limit_write respectively

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    s3cmd_reusable.rate_limit_read(bucket_name, max_read_bytes_kb, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+    s3cmd_reusable.rate_limit_write(bucket_name, max_write_bytes_kb, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    # Set the rate limits for the user and enable them
```
Comment (mkasturi18) [inline rgw/v2/tests/s3cmd/test_ratelimit_global.py:130]: Can setting rate limit at user level and enable  be written as a method in reusable

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+    max_read_bytes_kb = math.ceil(float(max_read_bytes) / 1024)
+    max_write_bytes_kb = math.ceil(float(max_write_bytes) / 1024)
+
+    # create bucket and set limits
+    bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=0)
+
+    ssl = config.ssl
+    s3cmd_reusable.create_bucket(bucket_name, ssl)
+    log.info(f"Bucket {bucket_name} created")
```
Comment (mkasturi18) [inline rgw/v2/tests/s3cmd/test_ratelimit_global.py:85]: @TejasC88 , can we have a method to enable limits at bucket level.

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    )
+    log.info(f"Rate limits enabled globally on bucket : {limget} ")
+
+    # test the read and write ops limit
+    log.info("Testing global read and write ops limit")
+    s3cmd_reusable.rate_limit_read(bucket_name, max_read_ops, ssl)
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
```
Comment (mkasturi18) [inline rgw/v2/tests/s3cmd/test_ratelimit_global.py:112]: @TejasC88 , can we add the sleep duration in the method rate_limit_read rate_limit_write respectively

File: rgw/v2/tests/s3cmd/test_bucket_stats.py (hunk 1 above)
Comment (anrao19) [reply rgw/v2/tests/s3cmd/test_bucket_stats.py]: @TejasC88 , as per the log: http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/bucket_stats_s3cmd.txt
we are not observing the issue mentioned in : https://bugzilla.redhat.com/show_bug.cgi?id=2181424

looks like its differnt issue .

@viduship , yes the bug is fixed. we are no longer observing this issue

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py (hunk 2 above)
Comment (TejasC88) [reply rgw/v2/tests/s3cmd/test_ratelimit_global.py]: @mkasturi18 Changed

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py (hunk 5 above)
Comment (TejasC88) [reply rgw/v2/tests/s3cmd/test_ratelimit_global.py]: changed

File: rgw/v2/tests/s3cmd/test_ratelimit_global.py (hunk 3 above)
Comment (TejasC88) [reply rgw/v2/tests/s3cmd/test_ratelimit_global.py:118]: @mkasturi18 Added a method to set and enable limit

**Review by mkasturi18 — APPROVED** [review]
changes looks good

**Pull request merged**